src/coordinates_photo.py

```diff
@@ -177,38 +180,160 @@ def polar_to_cartesian(altitude, azimuth):
 
-    # Convert Cartesian coordinates to GPS coordinates 
-    λ = math.atan2(y, x)
-    φ = math.atan2(z, math.sqrt(x**2 + y**2))
+    # en rad
+    object_lat, object_lon = cartesian_to_lon_lat(x,y)
+    object_lat, object_lon = math.degrees(object_lat) , math.degrees(object_lon)
+    # print(f"lon: {lon} rad / {math.degrees(lon)}deg, lat: {lat} rad/{math.degrees(lat)}deg")
 
-    object_lon = math.degrees(λ)
-    object_lat = math.degrees(φ)
+    # object_lat, object_lon = method1(alt_az.alt.rad,alt_az.az.rad)
 
     return object_lat, object_lon
 
+def altaz_to_polar(alt, az):
+    """
+    Submethod to convert altitude (radians), azimuth (radians) of an object to polar coordinates (r,theta)
+    Returns
+    -------
+    (r, theta) : tuple
+    """
+    alt_objet = 10000 # hypothesis: flying at 10km altitude
+    r = math.cos(alt) * alt_objet # en mètres
+    theta = az
+    return r, theta
+
+def polar_to_cartesian(r, theta):
+    """
+    Submethod to convert radius (radians), theta (radians) of an object 
+    to cartesian coordinates (x,y,z) in meters
+    Returns
+    -------
+    (x, y, z) : tuple
+    """
+    x = r * math.sin(theta) # axis: 
+    y = r * math.cos(theta)
+
+    return x,y
+
+def cartesian_to_lon_lat(x,y):
+    """
+    Submethod to convert cartesian coordinates (x,y,z) (we assume z = 10km) 
+    to longitude, latitude in radians
+    Returns
+    -------
+    (lon, lat) : tuple
+    """
+    z = 10000 # en m
+    lon = math.atan(y/x)
+    lat = math.atan(z/(math.sqrt(x*x+y*y)))
+    return lon, lat
+
+def get_celestial_coordinates(x,y,path):
+    """
+    Submethod to compute celestial coordinates of a given pixel x,y in the image 
+    Returns 
+    -------
+    (Ra,Dec) : tuple
+    """
+
+    ref_pixel,ref_celestial,matrix = get_calibrated_camera_parameters(path)
+
+    delta_Ra = matrix[0][0] * (x - ref_pixel[0]) + matrix[0][1] * (y - ref_pixel[1])
+    delta_Dec = matrix[1][0] * (x - ref_pixel[0]) + matrix[1][1] * (y - ref_pixel[1])
+
+    Ra = ref_celestial[0] + delta_Ra
+    Dec = ref_celestial[1] + delta_Dec
+
+    return Ra,Dec
 
 
 def main():
 
-    # observer_location = (35.6895, 51.3896, 0)
-    # observer_time = "2021-06-21 00:00:00"
-    # object_coordinates = (0, 0)
-    # object_location = convert_coordinates(observer_location, observer_time, object_coordinates)
-    # print(f"The GPS coordinates of the object are: {object_location}")
+    # Define the expected format
+    expected_format = r"-?\d+(\.\d+)?,\s*-?\d+(\.\d+)?,\s*-?\d+(\.\d+)?"
+    
+    # Prompting the user for observer location
+    observer_location_input = input("Enter observer location (latitude, longitude, altitude): ")
+
+    # Check if input matches the expected format
+    if re.match(expected_format, observer_location_input):
+        print("Input is in the correct format.")
+    else:
+        print("Input is not in the correct format. Please enter in latitude, longitude, altitude format.")
+        return -1
```

Comment:
<optional> Instead of returning, shouldn't we loop until the input fits the format?